daily_read/__main__.py

```diff
@@ -93,20 +93,20 @@ def generate_all(upload=False, develop=False):
     for owner in modified_orders:
         if upload:
             report = daily_rep.populate_and_write_report(owner, modified_orders[owner], config_values.STATUS_PRIORITY)
-            # Publish reports
-            for status in modified_orders[owner]["projects"].keys():
-                for project in modified_orders[owner]["projects"][status]:
-                    uploaded = False
-                    uploaded = op.upload_report_to_order_portal(report, project, "published")
-                    if uploaded:
-                        op.projects_data.stage_data_for_project(project)
-            # Hide old reports
-            for status in modified_orders[owner]["delete_report_for"].keys():
-                for project in modified_orders[owner]["delete_report_for"][status]:
-                    uploaded = False
-                    uploaded = op.upload_report_to_order_portal("", project, "review")
-                    if uploaded:
-                        op.projects_data.stage_data_for_project(project)
+            # Publish reports with published, hide reports with review
+            for upload_category, report_state in {"projects": "published", "delete_report_for": "review"}.items():
+                for status in modified_orders[owner][upload_category].keys():
+                    for project in modified_orders[owner][upload_category][status]:
+                        uploaded = False
+                        report_upload = report if upload_category == "projects" else ""
+                        try:
+                            uploaded = op.upload_report_to_order_portal(report_upload, project, report_state)
```

Comment:
I think the log message inside this function will still say that the report was updated even if the report is empty string, so perhaps we could update that log message to be more clear when it's the emtpy string?

Reply:
Updated the message in `order_portal.py`